src/IO/ReadHelpers.cpp

```diff
@@ -530,25 +536,39 @@ void readEscapedStringIntoImpl(Vector & s, ReadBuffer & buf)
                 }
             }
         }
+
+        if (*buf.position() == '\r')
+            ++buf.position();
     }
 }
 
-template <typename Vector>
+template <typename Vector, bool support_crlf>
 void readEscapedStringInto(Vector & s, ReadBuffer & buf)
 {
-    readEscapedStringIntoImpl<Vector, true>(s, buf);
+    readEscapedStringIntoImpl<Vector, true, support_crlf>(s, buf);
 }
 
 
 void readEscapedString(String & s, ReadBuffer & buf)
 {
     s.clear();
-    readEscapedStringInto(s, buf);
+    readEscapedStringInto<String,false>(s, buf);
+}
+
+template<bool support_crlf>
+void readEscapedStringCRLF(String & s, ReadBuffer & buf)
```

Comment:
I think `readEscapedStringCRLF` function should not be template and always use `support_crlf=true`. This function is supposed to be used only in cases when we need support for CRLF, in other cases `readEscapedString` function should be used